### ISSUE TITLE
[mysql] Use Incremental Snapshot Framework for Oracle CDC Connector

### ISSUE THREAD

**Issue by molsionmo** [ververica/flink-cdc-connectors#976]
**Is your feature request related to a problem? Please describe.**
Adapte Oracle CDC Connector to the abstraction framework of https://github.com/ververica/flink-cdc-connectors/issues/724

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Comment by molsionmo** [comment]
Hi @leonardBang , i am a contributor of flink, please assign me to this issue, , I'll help to adapt the oracle CDC Connector to the Incremental Snapshot Framework.

**Comment by leonardBang** [comment]
Thanks @molsionmo for taking this ticket , I've assigned this to you.

**Comment by fuyun2024** [comment]
Hi @leonardBang @molsionmo , I'm very interested in this issue. Can I take part in this work together?

**Comment by leonardBang** [comment]
@molsionmo @fuyun2024 I think adapt Oracle CDC to Incremental Snapshot Framework is a big umbrella task which we can split it into multiple subtasks and then both of you can take some subtasks, in this way we can finish the task easily.
WDYT？

**Comment by molsionmo** [comment]
@leonardBang  Yes,I can take apart several classes that need to be implemented and implement them separately. I'll break it down this week

**Comment by molsionmo** [comment]
@leonardBang 
Hi, Xu, as of today, I have completed the definition and implementation of Oracle RedoLogOffset, And OracleSnapshotSplit OracleRedoLogSplit definition of implementation, but also lack of events, reader, enumerator, states, and the definition of partial assigners and implementation.

Do you think I should submit this version first, or complete the rest of the definition before submitting.

I think it is best that events, reader, enumerator, assigners and states the definition of good supplement unified submit a version.I expect it to be this Friday at the latest.

**Comment by leonardBang** [comment]
@molsionmo It depends on your favor, both two choices make sense to me

**Comment by molsionmo** [comment]
@leonardBang Hi xu, I found that Debezium needs to be upgraded to 1.6.4.Final version, Debezium Oracle has this configuration: snapshot.locking.mode. What is Debezium upgrade strategy in Flink CDC project.

https://debezium.io/documentation/reference/1.6/connectors/oracle.html#oracle-property-snapshot-locking-mode

**Comment by fuyun2024** [comment]
@molsionmo Hi, I've seen this configuration. I understand that this configuration locks table schema changes during the snapshot. However, in Flink CDC, we can support table schema changes during snapshots. I think we can build on this version.

**Comment by molsionmo** [comment]
@fuyun2024 Our target is that All connector can reuse IncrementalSnapshot Algorithm which offers parallel-reading, non-lock,checkpoint feature. 

In Debezium 1.5.4.Final, it obtains an EXCLUSIVE MODE lock on each of the monitored tables to prevent structural changes from occurring during creation of the snapshot. Debezium holds the locks for only a short time and does not provide a lock-free configuration during creation of the snapshot. So we need to execute it [GRANT LOCK ANY TABLE TO c##dbzuser CONTAINER=ALL;] in database. DBA will not allow this in most cases and we can not say that we provide non-lock feature.

**Comment by fuyun2024** [comment]
@molsionmo  Hi, I just verified MySQL snapshot reading. In flick CDC, schema changes are not allowed during snapshot reads. My previous understanding is wrong, and your suggestion is correct.

**Comment by leonardBang** [comment]
Resolved in https://github.com/ververica/flink-cdc-connectors/pull/1079
